wrappers/rust/icicle-cuda-runtime/src/memory.rs

```diff
@@ -1,67 +1,115 @@
-use crate::bindings::{cudaFree, cudaMalloc, cudaMallocAsync, cudaMemcpy, cudaMemcpyAsync, cudaMemcpyKind};
+use crate::bindings::{
+    cudaFree, cudaMalloc, cudaMallocAsync, cudaMemPool_t, cudaMemcpy, cudaMemcpyAsync, cudaMemcpyKind,
+};
+use crate::device::{self, get_device};
 use crate::error::{CudaError, CudaResult, CudaResultWrap};
 use crate::stream::CudaStream;
 use std::mem::{size_of, MaybeUninit};
 use std::ops::{Index, IndexMut, Range, RangeFrom, RangeFull, RangeInclusive, RangeTo, RangeToInclusive};
 use std::os::raw::c_void;
 use std::slice::from_raw_parts_mut;
 
+const UNDEFINED_DEVICE: i32 = -1;
+
 pub enum HostOrDeviceSlice<'a, T> {
     Host(Vec<T>),
-    Device(&'a mut [T]),
+    Device(&'a mut [T], i32),
 }
 
 impl<'a, T> HostOrDeviceSlice<'a, T> {
+    pub fn set_device_id(&mut self, new_id: i32) {
```

Comment:
What's this function for? Can't find a usage anywhere in the code

Reply:
setter for device_id of slice - I set it during slice on-device allocation - so indeed not used

Reply:
removed as discussed